src/dashboard/tests/test_cas.py

```diff
@@ -0,0 +1,249 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+
+from django.conf import settings
+from django.contrib.auth.models import User
+from django.test import TestCase, RequestFactory
+from django.test.client import Client
+from django.urls import reverse
+import pytest
+
+try:
+    import mock
+except ImportError:
+    from unittest import mock
+
+from components import helpers
+from components.accounts.backends import CustomCASBackend
+from components.accounts.signals import _user_is_administrator
+
+TEST_CAS_USER = "casuser"
+TEST_CAS_ADMIN_ATTRIBUTE = "usertype"
+TEST_CAS_ADMIN_ATTRIBUTE_VALUE_POSITIVE = "admin"
+TEST_CAS_ADMIN_ATTRIBUTE_VALUE_NEGATIVE = "regular"
+
+TEST_CAS_ATTRIBUTES_STRING_POSITIVE = {
+    TEST_CAS_ADMIN_ATTRIBUTE: TEST_CAS_ADMIN_ATTRIBUTE_VALUE_POSITIVE
+}
+TEST_CAS_ATTRIBUTES_STRING_NEGATIVE = {
+    TEST_CAS_ADMIN_ATTRIBUTE: TEST_CAS_ADMIN_ATTRIBUTE_VALUE_NEGATIVE
+}
+TEST_CAS_ATTRIBUTES_LIST_POSITIVE = {
+    TEST_CAS_ADMIN_ATTRIBUTE: [
+        TEST_CAS_ADMIN_ATTRIBUTE_VALUE_POSITIVE,
+        "attribute1",
+        "attribute2",
+    ]
+}
+TEST_CAS_ATTRIBUTES_LIST_NEGATIVE = {
+    TEST_CAS_ADMIN_ATTRIBUTE: [
+        TEST_CAS_ADMIN_ATTRIBUTE_VALUE_NEGATIVE,
+        "attribute1",
+        "attribute2",
+    ]
+}
+
+
+def mock_verify(ticket, service):
+    user = TEST_CAS_USER
+    attributes = {
+        "ticket": ticket,
+        "service": service,
+        TEST_CAS_ADMIN_ATTRIBUTE: TEST_CAS_ADMIN_ATTRIBUTE_VALUE_NEGATIVE,
+    }
+    pgtiou = None
+    return user, attributes, pgtiou
+
+
+def mock_verify_superuser(ticket, service):
+    user = TEST_CAS_USER
+    attributes = {
+        "ticket": ticket,
+        "service": service,
+        TEST_CAS_ADMIN_ATTRIBUTE: TEST_CAS_ADMIN_ATTRIBUTE_VALUE_POSITIVE,
+    }
+    pgtiou = None
+    return user, attributes, pgtiou
+
+
+@pytest.mark.skipif(
+    not settings.CAS_AUTHENTICATION, reason="tests will only pass if CAS is enabled"
+)
```

Comment:
I agree it would be great to make the tests run regardless of settings.

I spent an hour or two trying different variations of overriding settings with the decorators described at https://docs.djangoproject.com/en/1.11/topics/testing/tools/#overriding-settings but haven't been able to get it to work. There is a warning in the Django docs that the changes may happen too late to take effect for some settings, and I see a similar comment [in our test settings](https://github.com/artefactual/archivematica/blob/5d46475461c49997464338adc2be538f4e66a70d/src/dashboard/src/settings/test.py#L67-L68). Did you have something in mind that I might not be thinking of?

Reply:
I see! We have all these tests that will only run if the corresponding toggle is enabled, but in practice we never do. It sounds like we'll need multiple runs, e.g. `pytest --ds=settings.test.cas tests/test_cas.py` or similar.